bin_sort/bin_sort.c

```diff
@@ -0,0 +1,49 @@
+/* Following algorithm sorts the input array in ascending order
+ * Time Complexity : O(n)-> Avg case and O(n^2)-> Worst Case
+ */
+#include <stdio.h>
+#include <stdlib.h>
+
+struct bucket {
+	int count;
+	float* value;
+};
+
+int comparator(const int* first, const int* second) {
+	float a = *first, b = *second;
+	int result = a - b;
+	return (0 < result) - (result < 0);
+}
+
+void bucket_sort(float array[], int n) {
+	struct bucket buckets[n];
+	int a, b, c, bi;
```

Comment:
This variable was introduced for better understanding purpose.Should i remove this variable ?